packages/world/test/IndexModule.t.sol

```diff
@@ -0,0 +1,164 @@
+    bytes32[] memory keysWithValue = ReverseMapping.get(targetTableId, world, keccak256(abi.encode(value1)));
+
+    // Assert that the list is correct
+    assertEq(keysWithValue.length, 1, "1");
+    assertEq(keysWithValue[0], key1, "2");
+
+    // Set a another key with the same value
+    world.setRecord(namespace, sourceFile, keyTuple2, abi.encodePacked(value1));
+
+    // Get the list of entities with value2 from the target table
+    keysWithValue = ReverseMapping.get(targetTableId, world, keccak256(abi.encode(value1)));
+
+    // Assert that the list is correct
+    assertEq(keysWithValue.length, 2);
+    assertEq(keysWithValue[0], key1, "3");
+    assertEq(keysWithValue[1], key2, "4");
+
+    // Change the value of the first key
+    uint256 value2 = 2;
+
+    world.setRecord(namespace, sourceFile, keyTuple1, abi.encodePacked(value2));
+
+    // Get the list of entities with value1 from the target table
+    keysWithValue = ReverseMapping.get(targetTableId, world, keccak256(abi.encode(value1)));
+
+    // Assert that the list is correct
+    assertEq(keysWithValue.length, 1, "5");
+    assertEq(keysWithValue[0], key2, "6");
+
+    // Get the list of entities with value2 from the target table
+    keysWithValue = ReverseMapping.get(targetTableId, world, keccak256(abi.encode(value2)));
+
+    // Assert that the list is correct
+    assertEq(keysWithValue.length, 1, "7");
+    assertEq(keysWithValue[0], key1, "8");
+
+    // Delete the first key
+    world.deleteRecord(namespace, sourceFile, keyTuple1);
+
+    // Get the list of entities with value2 from the target table
+    keysWithValue = ReverseMapping.get(targetTableId, world, keccak256(abi.encode(value2)));
+
+    // Assert that the list is correct
+    assertEq(keysWithValue.length, 0, "9");
+  }
+
+  function testSetField() public {
+    _installIndexModule();
+
+    // Set a value in the source table
+    uint256 value1 = 1;
+
+    world.setField(namespace, sourceFile, keyTuple1, 0, abi.encodePacked(value1));
+
+    // Get the list of entities with value1 from the target table
+    bytes32[] memory keysWithValue = ReverseMapping.get(targetTableId, world, keccak256(abi.encode(value1)));
+
+    // Assert that the list is correct
+    assertEq(keysWithValue.length, 1);
+    assertEq(keysWithValue[0], key1);
+
+    uint256 value2 = 2;
+
+    // Change the value using setField
+    world.setField(namespace, sourceFile, keyTuple1, 0, abi.encodePacked(value2));
+
+    // Get the list of entities with value1 from the target table
+    keysWithValue = ReverseMapping.get(targetTableId, world, keccak256(abi.encode(value1)));
+
+    // Assert that the list is correct
+    assertEq(keysWithValue.length, 0);
+
+    // Get the list of entities with value2 from the target table
+    keysWithValue = ReverseMapping.get(targetTableId, world, keccak256(abi.encode(value2)));
+
+    // Assert that the list is correct
+    assertEq(keysWithValue.length, 1);
+    assertEq(keysWithValue[0], key1);
+  }
```

Comment:
can we add some gas tests too?

Reply:
Yeah let's add some but don't look too close, the gas cost is gonna be bad. Will be useful for optimizing the implementation though.